src/App.tsx

```diff
@@ -1,23 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [countItems, setCountItems] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const startIndex = (currentPage - 1) * countItems;
+  const endIndex = startIndex + countItems;
+  const itemsPerPage = items.slice(startIndex, endIndex);
+
+  function paginate(pageNumber: number) {
+    setCurrentPage(pageNumber);
+  }
+
+  const previousPage = () => {
+    if (currentPage !== 1) {
+      setCurrentPage(currentPage - 1);
+    }
+  };
+
+  const nextPage = () => {
+    if (currentPage !== Math.ceil(items.length / countItems)) {
+      setCurrentPage(currentPage + 1);
+    }
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${itemsPerPage[0].split(' ')[1]} - ${itemsPerPage[itemsPerPage.length - 1].split(' ')[1]} of 42)`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            value={countItems}
+            onChange={(event) => {
```

Comment:
Consider moving this logic from jsx to a separate function
```suggestion
            onChange={onSelect}
```